src/components/OfflineSearch.ts

```diff
@@ -0,0 +1,174 @@
+import { Component, createElement } from "react";
+import { findDOMNode } from "react-dom";
+
+import * as dijitRegistry from "dijit/registry";
+
+import { Alert } from "./Alert";
+import "../ui/OfflineSearch.css";
+
+type SearchMethodOptions = "equals" | "contains";
+
+interface ListView extends mxui.widget._WidgetBase {
+    _datasource: {
+        _constraints: string | null;
+        _setSize: number;
+        atEnd: () => boolean;
+        _pageSize: number;
+    };
+    _loadMore: () => void;
+    _onLoad: () => void;
+    _renderData: () => void;
+    update: () => void;
+}
+
+interface OfflineSearchProps {
+    defaultQuery: string;
+    searchEntity: string;
+    searchAttribute: string;
+    showSearchBar: boolean;
+    targetGridName: string;
+    searchMethod: SearchMethodOptions;
+}
+
+interface OfflineSearchState {
+    alertMessage?: string;
+    buttonVisibility?: string;
```

Comment:
Better to just use a normal boolean for this... `showButton: boolean;`
That way, you can use classNames to control whether the `hidden` class is added or not on line 68.